services/frontend-service/src/legacy-ui/App/AuthContext.tsx

```diff
@@ -0,0 +1,126 @@
+        storeAuthStateInCookie: false,
+    },
+});
+
+const getLoginRequest = () => ({
+    scopes: ['User.Read'],
+});
+
+type AuthTokenContextType = {
+    token: String;
+    authHeader: {
+        Authorization?: String;
+    };
+};
+
+export const AuthTokenContext = React.createContext<AuthTokenContextType>({} as AuthTokenContextType);
+
+function AzureAuthTokenProvider({ children }: { children: React.ReactNode }): JSX.Element {
+    const loginRequest = React.useMemo(() => getLoginRequest(), []);
+    const [token, setToken] = React.useState('');
+    const [authHeader, setAuthHeader] = React.useState({});
+    const { instance, accounts } = useMsal();
+
+    React.useEffect(() => {
+        const request = {
+            ...loginRequest,
+            account: accounts[0],
+        };
+        instance
+            .acquireTokenSilent(request)
+            .then((response) => {
+                setToken(response.accessToken);
+                setAuthHeader({ Authorization: response.accessToken });
+            })
+            .catch(() => {
+                instance.acquireTokenPopup(request).then((response) => {
+                    setToken(response.accessToken);
+                    setAuthHeader({ Authorization: response.accessToken });
+                });
+            });
+    }, [instance, accounts, loginRequest]);
+    return <AuthTokenContext.Provider value={{ token, authHeader }}>{children}</AuthTokenContext.Provider>;
+}
+
+const AzureAutoSignIn = () => {
+    const isAuthenticated = useIsAuthenticated();
+    const loginRequest = React.useMemo(() => getLoginRequest(), []);
+    const { instance } = useMsal();
+    if (!isAuthenticated) {
+        instance.loginRedirect(loginRequest);
+    }
+    return <></>;
+};
+
+export function AuthProvider({ children }: { children: React.ReactNode }): JSX.Element {
+    const { configs } = React.useContext(ConfigsContext);
+    const useAzureAuth = configs?.authConfig?.azureAuth?.enabled || false;
+    const useAuth = useAzureAuth;
+    const msalConfig = React.useMemo(() => getMsalConfig(configs), [configs]);
+    const msalInstance = new PublicClientApplication(msalConfig);
+
+    return (
+        <>
+            {!!configs && Object.keys(configs).length > 0 && (
+                <AuthContext.Provider value={{ useAzureAuth, useAuth }}>
+                    {useAzureAuth ? (
+                        <MsalProvider instance={msalInstance}>
+                            <AuthenticatedTemplate>
+                                <AzureAuthTokenProvider>{children}</AzureAuthTokenProvider>
+                            </AuthenticatedTemplate>
+                            <UnauthenticatedTemplate>
+                                <AzureAutoSignIn />
+                            </UnauthenticatedTemplate>
+                        </MsalProvider>
+                    ) : (
+                        <AuthTokenContext.Provider value={{ token: '', authHeader: {} }}>
```

Comment:
Why wrapping `{children}` with Azure Auth, did you consider use cases that we will not use it?

Reply:
When auth is not used, the children will still need to use the hook (as hook shouldn't be called conditionally)
```
    const { authHeader } = React.useContext(AuthTokenContext);
```

So this removes the need for the children to conditionally call the hook (which isn't good). This way an empty hook is provided when there is no auth and there is no issue.